### ISSUE TITLE
Add `touchToSeek` property.

### ISSUE THREAD

**Issue by michalchudziak** [react-native-community/react-native-slider#211]
## Describe the Feature
We should add `touchToSeek` property to be able to toggle this functionality as it is not the default behavior for iOS, but also for better customisation.

## Possible Implementations

Add `touchToSeek` property (boolean, defaults to `true` on android and web and `false` to iOS).
Related code:
- iOS: https://github.com/react-native-community/react-native-slider/pull/117
- Android: native behavior, need to implement the opposite
- web: https://github.com/react-native-community/react-native-slider/pull/178

**Comment by ouabing** [comment]
I'm also looking forward to this property. I wrap the slider in a bottom sheet which can be toggle vertically. It's easy to trigger seeking accidentally when I want to scroll the bottom sheet.

**Comment by bewildergeist** [comment]
Agreed, this would be very welcome. I just recently upgraded to the version that includes the iOS tap handling from #201, and now our users constantly accidentally set sider values when scrolling through a list of filters that all use sliders 🙃

**Comment by smallsaucepan** [comment]
Hi @lukewalczak. Are you currently working on this? If not, I would be happy to have a crack as a first issue.

**Comment by lukewalczak** [comment]
Hey @smallsaucepan, feel free to take over the issue ✌🏼 If you don't mind, I have the solution for iOS so can push the changes on your feature branch, just let me know! Have a good day!

**Comment by smallsaucepan** [comment]
Sure thing @lukewalczak. Thank you for the head start! You can push your work in progress to https://github.com/smallsaucepan/react-native-slider/tree/feature/touch-to-seek-prop and I'll build on that.

Do @michalchudziak or @wojteg1337 need to add me as a contributor before you can reassign the issue?

It seems like one of the main drivers to be able to turn off this behaviour on iOS are problems related to scrolling. The example app exhibits different behaviour in this regard between iOS and Android (simulators only I'm afraid), so I'm wondering if there is also a problem with the way the tap handler is functioning?

On Android the slider value only changes on touch end *provided* minimal vertical scrolling took place since touch start. On iOS the slider value updates as soon as touch start has occurred and any movement takes place, including a vertical scrolling motion that includes a slight sideways delta. Do we need to rework the event handling on iOS as a first step?

![ezgif-3-68a299118706](https://user-images.githubusercontent.com/5403628/94324555-b6537700-ffdd-11ea-8bc9-9c9adc666864.gif)

**Comment by smallsaucepan** [comment]
Think I have a fix for this. Essentially lifted the implementation of the UITapGestureRecognizer into RNCSliderManager.

Would we expect a tap to fire a single onValueChange and a single onSlidingComplete event with the same value? If so, this will also address #212.
